repo/content/content_manager_lock_free.go

```diff
@@ -31,7 +31,10 @@ func (sm *SharedManager) maybeCompressAndEncryptDataForPacking(data gather.Bytes
 
 	// If the content is prefixed (which represents Kopia's own metadata as opposed to user data),
 	// and we're on V2 format or greater, enable internal compression even when not requested.
-	if contentID.HasPrefix() && comp == NoCompression && mp.IndexVersion >= index.Version2 {
+	// TODO(prasad): Get rid of complete block once metadata compression setting is implemented for
+	// all the prefixes
+	// For now, exclude this for `k` prefixed metadata.
+	if contentID.HasPrefix() && contentID.Prefix() != "k" && comp == NoCompression && mp.IndexVersion >= index.Version2 {
```

Comment:
What was the reasoning to skip compression for "k" in this PR ?

Reply:
NVM, I just re-read the code. This logic was overriding compression policy for dirs.